oolab/src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -10,16 +10,20 @@ public abstract class AbstractWorldMap implements IWorldMap, IPositionChangeObse
 
     @Override
     public void positionChanged(Vector2d oldPosition, Vector2d newPosition){
-        animals.put(newPosition, animals.get(oldPosition));
-        animals.remove(oldPosition);
+        Animal animal = animals.remove(oldPosition);
+        if (animal!= null) {
+            animals.put(newPosition, animal);
+        }
     }
 
     @Override
     public boolean place(Animal animal) {
```

Comment:
skoro rzucamy wyjątkiem to nie zwrócimy tu nigdy false -> ta metoda nie powinna już zwracać boolean